### PR TITLE
Breeding menu primary functionality

### PR DESCRIPTION
**Pull request by ashersummers**
Trying to get the breeding menu done. For testing purposes, made breeding menu launch at the start of the game instead of the traversal menu.

Draft below is what I'm aiming for:

breedingmenudraft

*As of 1/22 the main ui is in, titles for each sections show, a breed button shows, and the drone list on the right populates with living ally drones

### CODE REVIEW

**Comment by ashersummers** [comment]
new changes are that drones can be selected from the right part and used for breeding, which happens ingame before the breeding menu reopens. still a lot left to be tweaked and done so I think we should limit this PR to this stuff and assign out specific parts of it

File: Assets/Resources/Scripts/BreedingMenu.cs [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    public GameObject selectTwoContent;
+    private bool selectTwoSelecting;
+    public GameObject firstSelectedDrone;
+    public GameObject secondSelectedDrone;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        //Get list of living ally drones and populate scroll view with it
+        List<GameObject> droneList = GameObject.Find("Main Camera").GetComponent<Game>().livingAllies;
```
Comment (jacktmcdonnell) [inline Assets/Resources/Scripts/BreedingMenu.cs]: We are accessing this gamecomponent a lot through this script. It is very ineffecient to do `.find` every time on the main camera.

Should probably load the game component into memory with a private variable so we don't have to retrieve it a bunch of times.

```
private Game gameComponent;
    void Awake()
    {
        // Cache the Game component reference
        GameObject mainCamera = GameObject.Find("Main Camera");
        if (mainCamera != null)
        {
            gameComponent= mainCamera.GetComponent<Game>();
            if (gameComponent == null)
            {
                Debug.LogError("Game component not found on Main Camera.");
            }
        }
        else
        {
            Debug.LogError("Main Camera not found.");
        }
    }
```

then the start function would look like this

```
   void Start()
    {
        if (gameComponent != null)
        {
            InitializeDroneScrollView(gameComponent.livingAllies);
        }
        ResetSelection();
    }
```

File: Assets/Resources/Scripts/BreedingMenu.cs [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+        DisableSelection();
+
+        if (firstSelectedDrone == null || secondSelectedDrone == null) {
+            //TODO: Graceful handling for this situation
+            Debug.Log("Didn't have first or second drone selected when trying to breed");
+            return;
+        }
+
+        GameObject.Find("Main Camera").GetComponent<Game>().breedingTarget1 = firstSelectedDrone.GetComponent<DroneImage>().linkedDrone.GetComponent<Drone>();
```
Comment (jacktmcdonnell) [inline Assets/Resources/Scripts/BreedingMenu.cs]: Here can use 
````
gameComponent.breedingtarget1 = firstSelectedDrone.GetComponent<DroneImage>().linkedDrone.GetComponent<Drone>();
````

File: Assets/Resources/Scripts/BreedingMenu.cs (hunk 1 above)
Comment (ajm1996) [reply Assets/Resources/Scripts/BreedingMenu.cs]: there's a global variable for the main camera that I'm pretty sure has no overhead, that's what I use. it'd be ``Camera.main.getComponent<Game>();``. Either way it shouldn't be a huge deal if it's only called once per menu open.

File: Assets/Resources/Scripts/BreedingMenu.cs [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        GameObject.Find("Main Camera").GetComponent<Game>().breedingTarget2 = secondSelectedDrone.GetComponent<DroneImage>().linkedDrone.GetComponent<Drone>();
+        GameObject.Find("Main Camera").GetComponent<Game>().Breed();
```
Comment (ajm1996) [inline Assets/Resources/Scripts/BreedingMenu.cs]: it may be cleaner if I make the breeding targets an argument of the function rather than fields

**Review by ajm1996 — APPROVED** [review]

**Pull request merged**